src/overlays/actors/ovl_Bg_Hidan_Rsekizou/z_bg_hidan_rsekizou.c

```diff
@@ -21,13 +23,197 @@ const ActorInit Bg_Hidan_Rsekizou_InitVars = {
+            (this->dyna.actor.initPosRot.pos.z - yawSine * sphere->dim.modelSphere.center.x) +
+            yawCosine * sphere->dim.modelSphere.center.z;
+    }
+
+    CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_8002F974(&this->dyna.actor, NA_SE_EV_FIRE_PILLAR - SFX_FLAG);
+}
+
+Gfx* BgHidanRsekizou_DrawFireball(GlobalContext* globalCtx, BgHidanRsekizou* this, s16 frame, MtxF* mf, s32 a,
+                                  Gfx* displayList) {
+    f32 coss;
+    f32 sins;
+    s32 temp;
+    f32 fVar6;
+    f32 tmpf7;
+
+    temp = (((this->burnFrame + frame) % 8) * 7) * 0.14285715f;
+    gSPSegment(displayList++, 9, SEGMENTED_TO_VIRTUAL(D_8088CD74[temp]));
+
+    fVar6 = (++frame != 4) ? frame + ((3 - this->blastFrame) * 0.33333334f) : frame;
+
+    gDPSetPrimColor(displayList++, 0, 0x01, 255, 255, 0, 150);
+    gDPSetEnvColor(displayList++, 255, 0, 0, 255);
+
+    if (a == 0) {
+        sins = -Math_Sins(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+        coss = -Math_Coss(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+    } else {
+        sins = Math_Sins(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+        coss = Math_Coss(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+    }
+
+    mf->xx = mf->yy = mf->zz = (0.7f * fVar6) + 0.5f; /* [2][2] */
+    tmpf7 = (((((0.7f * fVar6) + 0.5f) * 10.0f) * fVar6) + 20.0f);
+
+    mf->wx = (tmpf7 * sins) + this->dyna.actor.posRot.pos.x;                      /* [3][0] */
+    mf->wy = (this->dyna.actor.posRot.pos.y + 30.0f) + (0.699999988079f * fVar6); /* [3][1] */
+    mf->wz = (tmpf7 * coss) + this->dyna.actor.posRot.pos.z;                      /* [3][2] */
+
+    gSPMatrix(displayList++,
+              Matrix_MtxFToMtx(Matrix_CheckFloats(mf, "../z_bg_hidan_rsekizou.c", 543),
+                               Graph_Alloc(globalCtx->state.gfxCtx, 0x40)),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(displayList++, D_0600DC30);
+
+    return displayList;
+}
+
+void BgHidanRsekizou_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanRsekizou* this = THIS;
+    s32 i;
+    s32 pad;
+    MtxF mf;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
+
+    gfxCtx = globalCtx->state.gfxCtx;
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_hidan_rsekizou.c", 564);
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_hidan_rsekizou.c", 568),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(gfxCtx->polyOpa.p++, &D_0600AD00);
+    Matrix_MtxFCopy(&mf, &gMtxFClear); /* Set an Identity Matrix */
+
+    gfxCtx->polyXlu.p = Gfx_CallSetupDL(gfxCtx->polyXlu.p, 0x14);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Rsekizou/BgHidanRsekizou_Update.s")
+    if ((s16)((func_8005A9F4(globalCtx->cameraPtrs[globalCtx->activeCamera]) - this->dyna.actor.shape.rot.y) -
```

Comment:
is the s16 cast needed here or can it be omitted?
Also, globalCtx->cameraPtrs[globalCtx->activeCamera] is a macro called ACTIVE_CAM

Reply:
This cast is definitely needed. Removing it causes the code to not match.

Reply:
Can you add a s16 temp for it? Instead of inlining it and casting it explicitely.
And the ACTIVE_CAM should be used here as Fig mentioned.

Reply:
Did you give a try to a s16 temp for this? I think you may have missed that part of my suggestion.

Reply:
Trying to make an s16 temp seemed to break the assembly, unless I was doing it wrong.

Reply:
was it just the stack that was wrong or instructions were wrong?
stack can probably be fixed

Reply:
yea i just messed around with it myself and it doesnt seem to work out, tried a few combinations of what goes in the temp